server/src/frameworks/Foundry/ImportCompleter.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  CompletionItem,
+  CompletionItemKind,
+  Position,
+} from "vscode-languageserver-types";
+import { SolFileIndexMap } from "../../parser/common/types";
+import { Remapping } from "./Remapping";
+
+export class ImportCompleter {
```

Comment:
True, it can be done both ways :sweat_smile: but this is the one that comes naturally to me, and I thought that it'd be appropiate since this part of the code uses classes as well due to the recent refactor. But I'm open to change the style of programming if you think it's more appropiate/desirable for this project. In any case, I'd like to stick to just one approach. Either using mostly classes, or all plain objects with functions that operate on them.

Reply:
I think that is a false dichotomy; we can leverage classes and have have standalone functions in one code base.

What we need is a convention on when we would use a class and when we would use a standalone function.

The framework adapter is a good example of where a class is justified (multiple implementations to same interface, need to combine state + logic). It pays for itself as an abstraction.

The `ImportCompleter` doesn't need inheritance, different implementations aren't going to be used from within the Foundry adapter, the state doesn't have a different lifetime from individual function calls; it is just a holder for the `ImportComplete` process.

Reply:
Makes sense. I'll try to better distinguish when to use which :+1: